src/microbe_stage/editor/OrganellePopupMenu.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Globalization;
+using Godot;
+
+/// <summary>
+///   Manages a custom context menu solely for showing list of options for a placed organelle
+///   in the microbe editor.
+/// </summary>
+public class OrganellePopupMenu : PopupPanel
+{
+    [Export]
+    public NodePath SelectedOrganelleNameLabelPath;
+
+    [Export]
+    public NodePath DeleteButtonPath;
+
+    [Export]
+    public NodePath MoveButtonPath;
+
+    private Label selectedOrganelleNameLabel;
+    private Button deleteButton;
+    private Button moveButton;
+
+    private bool showPopup;
+    private OrganelleTemplate selectedOrganelle;
+    private bool enableDelete = true;
+    private bool enableMove = true;
+
+    [Signal]
+    public delegate void DeletePressed();
+
+    [Signal]
+    public delegate void MovePressed();
+
+    public bool ShowPopup
+    {
+        get => showPopup;
+        set
+        {
+            showPopup = value;
+
+            if (ShowPopup)
+            {
+                RectPosition = GetViewport().GetMousePosition();
+                ShowModal();
```

Comment:
I think this should link with a TODO comment to https://github.com/Revolutionary-Games/Thrive/issues/1857